Operators.js

```diff
@@ -0,0 +1,66 @@
+console.log("Operators");
+// 1- Mathematically, if a == b and b == c, then a == c. Can you explain in your words why this is not true for the following snippet?
+
+//  0 == '0'; -> true
+//  0 == [];  -> true
+// "0" == []; -> false
+
+// Explanation: 
+//   - Equal (==) returns true if the operands are equal. it's compare the values.
+//   - Every value has an inherent boolean value(truthy or falsy).
+//      
+//      *First Line: 
+//       -  leftside is number and the value is 0 
+//       -  Rightside is string but javascript convert it to number (which is 0)                          
+//       -  So now both sides are numbers. As result the values are equal which returns (true)  
+
+//      *Second line:
+//       -  The empty array has inherited boolean value which is truthy.
+//       -  Now the equality operator (which is implicit coercion ) wants to compare [] with number
+//       -  The loose equality operator (==)use .toString() method to coerce empty array to an empty string which is coerced to zero 
+//       -  In the leftside, we have zero
+//          ->  0 == 0 return true
+//       
```

Comment:
*Second line:
//       -  The empty array has inherited boolean value which is truthy.
In console I get this:
[] == false
true
so empty array is not coerced to an empty string but to boolean value false. 0 == false so that's why this statement is true

Reply:
Hi Marc; 
Thanks for your feedback. I read in the website the you mentiond (https://www.freecodecamp.org/news/explaining-the-best-javascript-meme-i-have-ever-seen/) the following :

> Because coercion runs twice:
> ToPrimitive([]) gives empty string
> Then ToNumber("") gives 0.

this almost corresponds with what i have mentioned,  so i didn't get your point. could you please explian me the different. 
thanx +1: